Automattic/wp-calypso
# Add e2e tests for Woo authentication flow

Related to https://github.com/woocommerce/team-ghidorah/issues/237

## Proposed Changes

There is insufficient automated test coverage around that part of the WooCommerce authentication flow. Lately, we've been seeing changes that break the flow, such as https://github.com/woocommerce/team-ghidorah/issues/237.

This PR adds e2e tests below for Woo signup and login flow to improve the test coverage so that we can catch errors as early as possible.

| Test | Test Group | Frequency |
|--------|--------|--------|
| Email Signup | calypso-release | occur once a PR is approved and merged |
| Apple Signup | quarantined | once a day |
| Google Signup | calypso-release | occur once a PR is approved and merged |
| Username Login | authentication | once every 6 hours |
| Apple Login | authentication | once every 6 hours | 
| Google | authentication | once every 6 hours | 

Apple has a limit on the number of times we can request an OTP code within a period, so we're not able to run Apple login/signup tests frequently

## Testing Instructions



I've triggered Pre-Release / Authentication E2E / Quarantined E2E Tests for this PR on Team City. **All passed except for one flaky test in Quarantined E2E Tests that failed that was not related to this PR.**

To test locally:

* Checkout this branch
* Set up local e2e test env https://github.com/Automattic/wp-calypso/tree/trunk/test/e2e
* Run the following tests

- `yarn workspace wp-e2e-tests test test/e2e/specs/onboarding/signup__woo-email.ts`
- `yarn workspace wp-e2e-tests test test/e2e/specs/onboarding/signup__woo-apple.ts`
- `yarn workspace wp-e2e-tests test test/e2e/specs/onboarding/signup__woo-google.ts`
- `yarn workspace wp-e2e-tests test --group=authentication`

Confirmed all tests passed. (Please note that since the same OPT code can only be used once, the test will fail if someone else happens to test it at the same time.)


## Pre-merge Checklist
